kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/clients/KafkaClient.java

```diff
@@ -37,10 +40,9 @@ public interface KafkaClient {
      *
      * @param topicName the topic name
      * @param bootstrap the bootstrap
-     * @param message the message
      * @param numOfMessages the num of messages
      * @param timeout the timeout
-     * @return the string
+     * @return the list of ConsumerRecords
      */
-    String consumeMessages(String topicName, String bootstrap, String message, int numOfMessages, Duration timeout);
+    List<ConsumerRecord<String, String>> consumeMessages(String topicName, String bootstrap, int numOfMessages, Duration timeout);
```

Comment:
Judging by what you've had to do in the toConsumerRecord methods, I don't think I'd use Kafka's ConsumerRecord.  Let's declare our own ConsumerRecord interface with just fields that the important to the systems tests (key, value, headers,..).

Reply:
I don't think you've addressed this comment.

Reply:
Not yet

Reply:
I was thinking about this. Don't know if having own `ConsumerRecord` would simplify the code. We have to define all the JSON elements in each kafka client ConsumerRecord, if not the `ObjectMapper` fails. And the one who is causing more problems is the Header, as each kafka client reports the header in a different way, so the `toConsumerRecord` method would be pretty similar as we have right now. WDYT?

Reply:
My suggestion was to make ConsumerRecord an interface.  Just have getters exposing key, value etc.   Have three concrete implementations of the ConsumerRecord which know about the precise way the client represents the record. The systemtests only reference the ConsumerRecord.  The concrete impls of ConsumerRecord are private cocern.

Reply:
I don't think the interface should be in terms of org.apache.kafka.clients.consumer.ConsumerRecord.  It just seems to bring a whole load of concerns we don't really care about.